docs/src/tutorials/write-my-software-management-plugin.md

```diff
@@ -0,0 +1,374 @@
+This can be needed when an order of processing software modules is relevant.
+
+In other words, you can choose a combination of the `install` or `remove` commands or this `update-list` command up to your requirement.
+If you don't want to use `update-list`, the plugin must return `1` like the first _docker_ plugin example.
+
+```shell
+case "$COMMAND" in
+    ...
+    update-list)
+        exit 1
+        ;;
+esac
+```
+
+Let's expand the first _docker_ plugin example to use `update-list`.
+First, learn what is the input of `update-list`.
+
+The Software Management Agent calls a plugin as below:
+
+```shell
+$ myplugin update-list <<EOF
+install name1 version1
+install name2 "" path2
+remove "name 3" version3
+remove name4
+EOF
+```
+
+The point is that it doesn't take any command-line argument, 
+but the software action list is sent through **stdin**.
+
+The behaviour of operations `install` and `remove` is same as for original commands `install` and `remove` as [above](#install).
+
+That is equivalent to the use of original commands (`install` and `remove`):
+
+```shell
+$ myplugin install name1 --module-version version1
+$ myplugin install name2 --module-path path2
+$ myplugin remove "name 3" --module-version version3
+$ myplugin remove name4
+```
+
+To make the _docker_ plugin accept a list of install and remove actions,
+let's change the file as below.
+
+Filename: /etc/tedge/sm-plugins/docker
+
+```shell
+#!/bin/bash
+
+# Command-line argument is the only command type
+COMMAND="$1"
+
+read_module() {
+    if [ $# -lt 2 ]
+    then
+        echo "Missing version or path for sw-module '${1}'"
+    else
+        mOperation=${1}
+        shift
+        mName=${1}
+        shift
+        mVersion=${1}
+        shift
+        mPath=${1}
+        shift
+        echo "info: $mOperation, $mName, $mVersion, $mPath"
+        case "$mOperation" in
+           install)
+            sudo docker pull $mName:$mVersion
+            ;;
+          remove)
+            sudo docker rmi $mName:$mVersion
+            ;;
+        esac
+    fi
+}
+
+case "$COMMAND" in
+    list)
+        sudo docker image list --format '{"name":"{{.Repository}}","version":"{{.Tag}}"}'
+        ;;
+    install)
+        # We use update-list instead
+        ;;
+    remove)
+        # We use update-list instead
+        ;;
+    prepare)
+        ;;
+    finalize)
+        ;;
+    update-list)
+        echo "---+++ reading software list +++---"
+        while read -r line; do
+            eval "moduleArray=($line)";
+            read_module "${moduleArray[@]}"
+        done
+        ;;
+esac
+exit 0
+```
+
+You can find that `install` and `remove` are replaced by `update-list`.
+`update-list` should define the behaviour to read line by line for the case `install` and `remove`.
+
+## Project references
+
+**thin-edge.io** provides APT plugin written in Rust.
+You can check out the code from [here](https://github.com/thin-edge/thin-edge.io/tree/main/sm/plugins/tedge_apt_plugin).
```

Comment:
When available (soon !), it would be good to also add a link to the docker plugin.